src/main/jbake/content/blog/2017/07/jaeger-with-security-proxy.adoc

```diff
@@ -0,0 +1,252 @@
+like link:https://github.com/minishift/minishift[`minishift`]. If you are running a recent version of Fedora, CentOS or
+Red Hat Enterprise Linux you might want to install the package `origin-clients` and run `oc cluster up --version=latest`.
+This should get you a basic OpenShift cluster running locally.
+
+To make it easier for our demonstration, we'll add `cluster-admin` rights to our `developer` user and we'll create 
+the Jaeger namespace:
+
+```bash
+oc login -u system:admin
+oc new-project jaeger
+oc adm policy add-cluster-role-to-user cluster-admin developer
+oc login -u developer
+```
+
+=== Preparing the Jaeger OpenShift template
+
+We'll use the link:https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml[Jaeger OpenShift Production template]
+as the starting point: either clone the entire repository, or just get a local version of the template.
+
+The first step is to add the sidecar container to the `query-deployment` object. Under the `containers` list, after we
+specify the `jaeger-query`, let's add the sidecar:
+
+```yml
+        - image: jpkroehling/keycloak-proxy
+          name: ${JAEGER_SERVICE_NAME}-query-security-proxy
+          volumeMounts:
+          - mountPath: /opt/jboss/conf
+            name: security-proxy-configuration-volume
+          ports:
+          - containerPort: 8080
+            protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: "/"
+              port: 8080
+```
+
+Note that container specifies a `volumeMount` named `security-proxy-configuration-volume`: we'll use it to store the
+proxy's configuration file. The volume should be added under the `spec` node for the deployment's template and is a sibling
+node to the `dnsPolicy` property:
+
+```yml
+        volumes:
+          - configMap:
+              name: ${JAEGER_SERVICE_NAME}-configuration
+              items:
+                - key: proxy
+                  path: proxy.json
+            name: security-proxy-configuration-volume
+```
+
+Now, we need to specify the `ConfigMap`, with the proxy's configuration entry. To do that, we add a new top-level item to
+the template:
+
+```yml
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: ${JAEGER_SERVICE_NAME}-configuration
+    labels:
+      app: jaeger
+      jaeger-infra: security-proxy-configuration
+  data:
+    proxy: |
+      {
+          "target-url": "http://localhost:16686",
+          "bind-address": "0.0.0.0",
+          "http-port": "8080",
+          "applications": [
+              {
+                  "base-path": "/",
+                  "adapter-config": {
+                    "realm": "jaeger",
+                    "auth-server-url": "${REDHAT_SSO_URL}",
+                    "ssl-required": "external",
+                    "resource": "proxy-jaeger",
+                    "credentials": {
+                      "secret": "THE-SECRET-FROM-INSTALLATION-FILE"
```

Comment:
This is something that would be hard to debug if it gets the wrong value, specially if it looks "correct". So, if a reader copies/paste this directly, without paying enough attention, it might take a while to figure out the problem.